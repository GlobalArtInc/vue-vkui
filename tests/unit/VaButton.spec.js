import { mount } from '@vue/test-utils'
import VaButton from '@/Button/VaButton.vue'
import VaIcon from '@/Icon/VaIcon.vue'
import Vue from 'vue'

describe('VaButton.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(VaButton)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders without warnings', () => {
    let spy = jest.spyOn(console, 'error')
    mount(VaButton)
    expect(spy).toBeCalledTimes(0)
    spy.mockReset()
  })

  it('renders an anchor', () => {
    const wrapper = mount(VaButton)
    expect(wrapper.contains('a')).toBe(true)
  })

  it('accepts "primary" as prop "type"', () => {
    const wrapper = mount(VaButton, {
      propsData: {
        type: 'primary'
      }
    })
    expect(wrapper.props().type).toBe('primary')
  })

  it('should support disabled', () => {
    const wrapper = mount(VaButton, {
      propsData: {
        disabled: true
      }
    })
    expect(wrapper.classes()).toContain('va-btn-disabled')
  })

  it('prop "type" with value primary results in element with va-btn-primary class', () => {
    const wrapper = mount(VaButton, {
      propsData: {
        type: 'primary'
      }
    })
    expect(wrapper.classes()).toContain('va-btn-primary')
  })

  it('"default" should be the default value for prop "type"', () => {
    const wrapper = mount(VaButton)
    expect(wrapper.props().type).toBe('default')
  })

  it('values passed to slots should render correctly', () => {
    const wrapper = mount(VaButton, {
      slots: {
        default: '<span>Hello</span>'
      }
    })
    expect(wrapper.find('span').text()).toBe('Hello')
  })

  it('icon passed to slot should render correctly', () => {
    const wrapper = mount({
      render () {
        return (
          <VaButton>
            <VaIcon type='home' />
          </VaButton>
        )
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('value passed to iconBefore should render correctly', () => {
    const wrapper = mount({
      render () {
        return (
          <VaButton iconBefore="home">Hello</VaButton>
        )
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('value passed to iconAfter should render correctly', () => {
    const wrapper = mount({
      render () {
        return (
          <VaButton iconAfter="home">Hello</VaButton>
        )
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('should change loading state on click', done => {
    // prevent timeout
    Vue.config.errorHandler = done
    const AButton = {
      data () {
        return {
          loading: false
        }
      },
      methods: {
        enableLoading () {
          this.loading = true
        }
      },
      render () {
        return <VaButton loading={this.loading} onClick={this.enableLoading}>Hello</VaButton>
      }
    }
    const wrapper = mount(AButton)
    wrapper.trigger('click')
    Vue.nextTick(() => {
      expect(wrapper.findAll('va-loading').length).toBe(1)
      done()
    })
    expect(wrapper).toMatchSnapshot()
  })
})
