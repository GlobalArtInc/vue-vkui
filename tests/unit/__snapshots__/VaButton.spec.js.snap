// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaButton.vue icon passed to slot should render correctly 1`] = `
<a tabindex="0" class="va-btn va-btn-md va-btn-default">
  <div class="va-btn-text-fade">
    <!----> <i class="fas fa-home" style="font-size: 1em; margin: 0px; background-color: transparent; border-radius: 4px; padding: 0px;"></i>
    <!---->
  </div>
  <!---->
</a>
`;

exports[`VaButton.vue renders correctly 1`] = `
<a
  class="va-btn va-btn-md va-btn-default"
  tabindex="0"
>
  <div
    class="va-btn-text-fade"
  >
    <!---->
      
    <!---->
  </div>
   
  <!---->
</a>
`;

exports[`VaButton.vue should change loading state on click 1`] = `
<a tabindex="0" class="va-btn va-btn-md va-btn-default" style="width: 0px; height: 0px;">
  <div class="va-btn-text-fade va-btn-text-fade-out">
    <!----> Hello
    <!---->
  </div>
  <va-loading color="#45526B" size="md"></va-loading>
</a>
`;

exports[`VaButton.vue value passed to iconAfter should render correctly 1`] = `
<a tabindex="0" class="va-btn va-btn-md va-btn-default">
  <div class="va-btn-text-fade">
    <!----> Hello <va-icon type="home" style="position: absolute; right: 3px;"></va-icon>
  </div>
  <!---->
</a>
`;

exports[`VaButton.vue value passed to iconBefore should render correctly 1`] = `
<a tabindex="0" class="va-btn va-btn-md va-btn-default">
  <div class="va-btn-text-fade">
    <va-icon type="home" style="position: absolute; left: 3px;"></va-icon> Hello
    <!---->
  </div>
  <!---->
</a>
`;
